### PR TITLE
EPMRPP-83572 || Update the view of Notification rule with attributes

### PR DESCRIPTION
**Pull request by Vadim73i**

### CODE REVIEW

File: app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
 import { defineMessages, useIntl } from 'react-intl';
 import { AttributeListContainer } from 'components/containers/attributeListContainer';
+import { ATTRIBUTES_OPERATORS } from 'pages/common/settingsPage/notificationsTab/constants';
```
Comment (AmsterGet) [inline app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx:22]: Please use corresponding constants from the redesigned components (import from '../../notifications/constants' as below).

File: app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx [hunk 2]
```diff
@@ -102,7 +111,15 @@ export const NotificationRuleContent = ({ item }) => {
...
         <>
-          <span className={cx('field')}>{formatMessage(messages.attributesLabel)}</span>
+          <span className={cx('field', 'attributes-text')}>
+            {item.attributes.length > 1
+              ? `${formatMessage(messages.attributesLabel)} ${
+                  item.attributesOperator === ATTRIBUTES_OPERATORS.AND
+                    ? formatMessage(messages.attributesLabelAll)
+                    : formatMessage(messages.attributesLabelAny)
+                }`
+              : formatMessage(messages.attributesLabel)}
```
Comment (AmsterGet) [inline app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx:121]: Please move to a variable or function to simplify reading.

File: app/src/components/containers/AttributeListFormField/attributeListFormField.jsx [hunk 3]
```diff
@@ -54,10 +54,15 @@ export const AttributeListFormField = ({
...
 
-  const attributesCaption =
-    shown || (!shown && !value?.length)
-      ? attributesNote || formatMessage(messages.attributesNote)
-      : formatMessage(messages.attributesNotActive);
+  const getAttributesCaption = () => {
+    if (!shown) {
+      return value?.length
+        ? formatMessage(messages.attributesNotActive)
+        : attributesNote || formatMessage(messages.attributesNote);
```
Comment (AmsterGet) [inline app/src/components/containers/AttributeListFormField/attributeListFormField.jsx:61]: Looks like we'll provide custom `attributesNote` for all cases except the notifications tab. And if it is not provided, we'll face the bug that notifications related message shown for not only notifications related use cases.
So please lets move this specific message to the parent component and provide it via `attributesNote` prop.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx [hunk 4]
```diff
@@ -82,11 +81,12 @@ const messages = defineMessages({
 });
 
 export const NotificationRuleContent = ({ item }) => {
+  const { informOwner, recipients, attributes, attributesOperator, launchNames, sendCase } = item;
```
Comment (Bam6ycha) [inline app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx:84]: Looks like we can destructure it directly in func arguments

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx [hunk 5]
```diff
@@ -88,23 +98,32 @@ export const NotificationRuleContent = ({ item }) => {
     [LAUNCH_CASES.TO_INVESTIGATE]: formatMessage(messages[LAUNCH_CASES.TO_INVESTIGATE]),
   };
 
+  const attributesFieldText =
+    attributes.length > 1
+      ? `${formatMessage(messages.attributesLabel)} ${
+          attributesOperator === ATTRIBUTES_OPERATORS.AND
+            ? formatMessage(messages.attributesLabelAll)
+            : formatMessage(messages.attributesLabelAny)
```
Comment (AmsterGet) [inline app/src/pages/inside/projectSettingsPageContainer/content/elements/notificationRuleContent/notificationRuleContent.jsx:106]: `formatMessage(attributesOperator === ATTRIBUTES_OPERATORS.AND ? messages.attributesLabelAll : messages.attributesLabelAny)`

**Review by AmsterGet — CHANGES_REQUESTED** [review]
Please also fix SonarCloud issue

image

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**